lib/types.nix

```diff
@@ -94,9 +94,12 @@ rec {
     , # The deprecation message to display when this type is used by an option
       # If null, the type isn't deprecated
       deprecationMessage ? null
+    , # The subtypes of this type. This is currently only used for being able to
+      # issue deprecation warnings recursively
+      subtypes ? {}
```

Comment:
A subtype relation is typically defined by the property that all inhabitants of one type are inhabitants of some other type. Object-oriented programmers may think of the Liskov principle. In Haskell you have `a -> a` as a subtype of `Int -> Int`, etc.

I propose `occurringTypes`. `member`, `element` etc seems to be too specific. 'Occurrence' is used in the literature. 'Occurs check' comes to mind.

Reply:
I opted for `nestedTypes`, which I think is easier to understand.